File: dotnet/Lexy/Lexy.Compiler/Language/ScenarioFunctionName.cs
using System.Collections.Generic;
using Lexy.Compiler.Parser;

namespace Lexy.Compiler.Language
{
    public class ScenarioFunctionName : Node
    {
        public string Value { get; private set; }

        public INode Parse(IParserContext context)
        {
            var line = context.CurrentLine;
            Value = line.Tokens.TokenValue(1);
            return this;
        }

        public ScenarioFunctionName(SourceReference reference) : base(reference)
        {
        }

        public override string ToString() => Value;

        public override IEnumerable<INode> GetChildren()
        {
            yield break;
        }

        protected override void Validate(IValidationContext context)
        {
        }

        public bool IsEmpty() => string.IsNullOrEmpty(Value);
    }
}

File: typescipt/src/language/scenarios/scenarioFunctionName.ts


export class ScenarioFunctionName extends Node {
   public string Value { get; private set; }

   public ScenarioFunctionName(SourceReference reference) {
     super(reference);
   }

   public parse(context: IParseLineContext): void {
     let line = context.line;
     Value = line.tokens.tokenValue(1);
   }

   public override toString(): string {
     return Value;
   }

   public override getChildren(): Array<INode> {
     yield break;
   }

   protected override validate(context: IValidationContext): void {
   }

   public isEmpty(): boolean {
     return string.IsNullOrEmpty(Value);
   }
}
